From: Franck Bui <fbui@mandriva.com>
Subject: BKL: provide unlocked_ioctl fops

Usage of BKL is not needed by ndiswrapper at all (see SVN trunk r2726).

Signed-off-by: Franck Bui <fbui@mandriva.com>
---

diff -Nurp ndiswrapper-1.56.orig/driver/loader.c ndiswrapper-1.56/driver/loader.c
--- ndiswrapper-1.56.orig/driver/loader.c	2010-02-11 04:09:32.000000000 +0200
+++ ndiswrapper-1.56/driver/loader.c	2010-10-02 18:57:56.971689617 +0300
@@ -750,7 +753,11 @@ struct wrap_device *get_wrap_device(void
 }
 
 /* called with loader_mutex is down */
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 35)
+static long wrapper_ioctl(struct file *file,
+#else
 static int wrapper_ioctl(struct inode *inode, struct file *file,
+#endif
 			 unsigned int cmd, unsigned long arg)
 {
 	struct load_driver *load_driver;
@@ -831,7 +844,11 @@ static int wrapper_ioctl_release(struct
 
 static struct file_operations wrapper_fops = {
 	.owner          = THIS_MODULE,
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 35)
+	.unlocked_ioctl = wrapper_ioctl,
+#else
 	.ioctl		= wrapper_ioctl,
+#endif
 	.release	= wrapper_ioctl_release,
 };
 
